### DISCUSSION TITLE
Is it possible to create paint app using egui?

### DISCUSSION THREAD

**Discussion by krulsaidme0w** [emilk/egui#1734] [General]
What tools should I use for this?

**Comment by ray-5050** [comment]
egui has a demo app that does some painting or drawing.  It might be a good place to start for whatever you want to do.  Your question is a bit non specific but if you look at the demo app, maybe you can ask a more specific question.  Of course, the major tool you'll need is rust.

**Reply by krulsaidme0w** [reply]
i need a possibility to draw like in paint, when i choose color and when i click on pixel, it becomes this color, i do not know, how to do this using egui, i allocated space for rect, but i do not know how to draw on it or get pixel positions and etc

**Reply by ray-5050** [reply]
The demo app I mentioned above does that.  The app is in egui/egui_demo_lib/src/demo/painting.rs
I've got it pulled out here, https://github.com/ray-5050/egui-painting-demo.git

**Reply by krulsaidme0w** [reply]
thanks a lot, that is what i was looking for <3
